### PR TITLE
Hot potato on fire

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

Makes hot potato deal heat damage while you holding it

**Media**

**Changelog**

:cl:
- add: hot potato now deals damage while you are holding it

### CODE REVIEW

**Comment by Slava0135** [comment]
added burn sound effect
https://user-images.githubusercontent.com/40753025/235728649-b05ea01d-1ffe-4d20-984e-6e4d8bee86c7.mp4

File: Content.Server/HotPotato/HotPotatoSystem.cs [hunk 1]
```diff
@@ -52,4 +59,16 @@ private void OnMeleeHit(EntityUid uid, HotPotatoComponent comp, MeleeHitEvent ar
...
+    {
+        var query = EntityQueryEnumerator<ActiveHotPotatoComponent, HotPotatoComponent>();
+        while (query.MoveNext(out var uid, out var _, out var potato))
+        {
+            if (_container.TryGetContainingContainer(uid, out var container))
+            {
+                _damageableSystem.TryChangeDamage(container.Owner, frameTime * potato.HoldingDamage, origin: uid);
+            }
+        }
+    }
```
Comment (metalgearsloth) [inline Content.Server/HotPotato/HotPotatoSystem.cs:73]: This should be its own component for holding an item doing damage, also it's not a good idea to spam damage events every tick, just tick every second or something.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Damage/Components/DamageOnHoldingComponent.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public DamageSpecifier Damage = new();
+
+    /// <summary>
+    /// Delay between damage events in seconds
+    /// </summary>
+    [DataField("interval"), ViewVariables(VVAccess.ReadWrite)]
+    public float Interval = 1f;
+
+    [ViewVariables]
+    public TimeSpan NextDamage = TimeSpan.Zero;
```
Comment (metalgearsloth) [inline Content.Server/Damage/Components/DamageOnHoldingComponent.cs:26]: TimeOffsetSerializer + datafield.

File: Content.Server/Damage/Systems/DamageOnHoldingSystem.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            if (!component.Enabled || component.NextDamage > _timing.CurTime)
+                continue;
+            if (_container.TryGetContainingContainer(uid, out var container))
+            {
+                _damageableSystem.TryChangeDamage(container.Owner, component.Damage, origin: uid);
+            }
+            component.NextDamage += TimeSpan.FromSeconds(component.Interval);
+            if (component.NextDamage < _timing.CurTime) // on first iteration or if component was disabled for long time
+                component.NextDamage = _timing.CurTime + TimeSpan.FromSeconds(component.Interval);
```
Comment (metalgearsloth) [inline Content.Server/Damage/Systems/DamageOnHoldingSystem.cs:44]: Set this on enabled.

File: Content.Server/Damage/Systems/DamageOnHoldingSystem.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DamageOnHoldingComponent, EntityUnpausedEvent>(OnUnpaused);
+    }
+
+    public void SetEnabled(EntityUid uid, bool enabled)
+    {
+        if (TryComp<DamageOnHoldingComponent>(uid, out var component))
+            component.Enabled = enabled;
+    }
```
Comment (metalgearsloth) [inline Content.Server/Damage/Systems/DamageOnHoldingSystem.cs:24]: This should be a resolve that takes in the component instead.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Damage is predicted so should be able to put this in shared and network it too I think.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**